frontend/src/metabase/admin/settings/components/EmbeddingSettings/types.ts

```diff
@@ -0,0 +1,8 @@
+import type { SettingKey, Settings } from "metabase-types/api";
+
+export interface AdminSettingComponentProps {
+  updateSetting: <K extends SettingKey>(
```

Comment:
this allows auto-completing the settings key, so no human error, also it will force the value to match the setting as well which is huge.